Pada tugas kali ini, kita akan mencoba untuk membuat model dari data training untuk mengklasifikasikan data_uji.

Kita ingin melihat dengan fitur yang dimiliki apakah penumpang tersebut akan selamat atau tidak.

Algoritma yang akan kita gunakan adalah **logistic regression**.


In [2]:
#import semua package/library pada sel ini, jika butuh memasukkan package/library baru maka run ulang sel ini
import pandas as pd
import sklearn.feature_selection as fs
import warnings
warnings.filterwarnings('ignore')

from sklearn.feature_selection import RFE
from sklearn.linear_model import LogisticRegression 
from sklearn.model_selection import train_test_split 
from sklearn import metrics

2. Data Collection


In [3]:
#panggil data kalian dan masukkan pada dua variabel data_mentah untuk train dan data_uji untuk data_uji

data_mentah = pd.read_csv('train.csv')
data_uji = pd.read_csv('data_uji.csv')

In [4]:
#tampilkan head dari data_mentah
display(data_mentah.head())

Age Cabin Embarked     Fare  \
0  22.0   NaN        S   7.2500   
1  38.0   C85        C  71.2833   
2  26.0   NaN        S   7.9250   
3  35.0  C123        S  53.1000   
4  35.0   NaN        S   8.0500   

                                                Name  Parch  PassengerId  \
0                            Braund, Mr. Owen Harris      0            1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...      0            2   
2                             Heikkinen, Miss. Laina      0            3   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)      0            4   
4                           Allen, Mr. William Henry      0            5   

   Pclass     Sex  SibSp  Survived            Ticket Title  Family_Size  
0       3    male      1       0.0         A/5 21171    Mr            1  
1       1  female      1       1.0          PC 17599   Mrs            1  
2       3  female      0       1.0  STON/O2. 3101282  Miss            0  
3       1  female      1       1.0            113803   Mrs            1  
4       3    male      0       0.0            373450    Mr            0

In [145]:
#tampilkan head dari data_mentah
#expected result

Age Cabin Embarked     Fare  \
0  22.0   NaN        S   7.2500   
1  38.0   C85        C  71.2833   
2  26.0   NaN        S   7.9250   
3  35.0  C123        S  53.1000   
4  35.0   NaN        S   8.0500   

                                                Name  Parch  PassengerId  \
0                            Braund, Mr. Owen Harris      0            1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...      0            2   
2                             Heikkinen, Miss. Laina      0            3   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)      0            4   
4                           Allen, Mr. William Henry      0            5   

   Pclass     Sex  SibSp  Survived            Ticket Title  Family_Size  
0       3    male      1       0.0         A/5 21171    Mr            1  
1       1  female      1       1.0          PC 17599   Mrs            1  
2       3  female      0       1.0  STON/O2. 3101282  Miss            0  
3       1  female      1       1.0            113803   Mrs            1  
4       3    male      0       0.0            373450    Mr            0

In [5]:
#tampilkan head dari data_mentah
display(data_uji.head())

Age  Cabin Embarked     Fare  \
0  34.5    NaN        Q   7.8292   
1  47.0    NaN        S   7.0000   
2  62.0    NaN        Q   9.6875   
3  27.0    NaN        S   8.6625   
4  22.0    NaN        S  12.2875   

                                           Name  Parch  PassengerId  Pclass  \
0                              Kelly, Mr. James      0          892       3   
1              Wilkes, Mrs. James (Ellen Needs)      0          893       3   
2                     Myles, Mr. Thomas Francis      0          894       2   
3                              Wirz, Mr. Albert      0          895       3   
4  Hirvonen, Mrs. Alexander (Helga E Lindqvist)      1          896       3   

      Sex  SibSp  Survived   Ticket Title  Family_Size  
0    male      0       NaN   330911    Mr            0  
1  female      1       NaN   363272   Mrs            1  
2    male      0       NaN   240276    Mr            0  
3    male      0       NaN   315154    Mr            0  
4  female      1       NaN  3101298   Mrs            2

In [146]:
#tampilkan head dari data_uji
#expected result

Age  Cabin Embarked     Fare  \
0  34.5    NaN        Q   7.8292   
1  47.0    NaN        S   7.0000   
2  62.0    NaN        Q   9.6875   
3  27.0    NaN        S   8.6625   
4  22.0    NaN        S  12.2875   

                                           Name  Parch  PassengerId  Pclass  \
0                              Kelly, Mr. James      0          892       3   
1              Wilkes, Mrs. James (Ellen Needs)      0          893       3   
2                     Myles, Mr. Thomas Francis      0          894       2   
3                              Wirz, Mr. Albert      0          895       3   
4  Hirvonen, Mrs. Alexander (Helga E Lindqvist)      1          896       3   

      Sex  SibSp  Survived   Ticket Title  Family_Size  
0    male      0       NaN   330911    Mr            0  
1  female      1       NaN   363272   Mrs            1  
2    male      0       NaN   240276    Mr            0  
3    male      0       NaN   315154    Mr            0  
4  female      1       NaN  3101298   Mrs            2

3. Data Preparation

Pada bagian ini kita akan memperbaiki nan value

In [6]:
# menghitung ada berapa jumlah nilai missing value yang terakomodasi oleh NaN dan melakukan analisis secara deskriptif
# untuk data_mentah
display(data_mentah.isnull().sum())
display(data_mentah.describe())

Age              0
Cabin          687
Embarked         0
Fare             0
Name             0
Parch            0
PassengerId      0
Pclass           0
Sex              0
SibSp            0
Survived         0
Ticket           0
Title            0
Family_Size      0
dtype: int64

Age        Fare       Parch  PassengerId      Pclass  \
count  891.000000  891.000000  891.000000   891.000000  891.000000   
mean    29.445196   32.204208    0.381594   446.000000    2.308642   
std     13.244896   49.693429    0.806057   257.353842    0.836071   
min      0.420000    0.000000    0.000000     1.000000    1.000000   
25%     22.000000    7.910400    0.000000   223.500000    2.000000   
50%     30.000000   14.454200    0.000000   446.000000    3.000000   
75%     35.500000   31.000000    0.000000   668.500000    3.000000   
max     80.000000  512.329200    6.000000   891.000000    3.000000   

            SibSp    Survived  Family_Size  
count  891.000000  891.000000   891.000000  
mean     0.523008    0.383838     0.904602  
std      1.102743    0.486592     1.613459  
min      0.000000    0.000000     0.000000  
25%      0.000000    0.000000     0.000000  
50%      0.000000    0.000000     0.000000  
75%      1.000000    1.000000     1.000000  
max      8.000000    1.000000    10.000000

In [7]:
#Check detail dan nilai NaN pada data_mentah. gunakan print untuk Nilai NaN
#expected result

In [8]:
# menghitung ada berapa jumlah nilai missing value yang terakomodasi oleh NaN dan melakukan analisis secara deskriptif
# untuk data_uji
display(data_uji.isnull().sum())
display(data_uji.describe())

Age             0
Cabin          10
Embarked        0
Fare            0
Name            0
Parch           0
PassengerId     0
Pclass          0
Sex             0
SibSp           0
Survived       10
Ticket          0
Title           0
Family_Size     0
dtype: int64

Age  Cabin       Fare      Parch  PassengerId     Pclass  \
count  10.000000    0.0  10.000000  10.000000     10.00000  10.000000   
mean   30.150000    NaN  12.270010   0.200000    896.50000   2.800000   
std    14.541225    NaN   7.777627   0.421637      3.02765   0.421637   
min    14.000000    NaN   7.000000   0.000000    892.00000   2.000000   
25%    21.250000    NaN   7.679200   0.000000    894.25000   3.000000   
50%    26.500000    NaN   8.943750   0.000000    896.50000   3.000000   
75%    33.375000    NaN  11.637500   0.000000    898.75000   3.000000   
max    62.000000    NaN  29.000000   1.000000    901.00000   3.000000   

           SibSp  Survived  Family_Size  
count  10.000000       0.0    10.000000  
mean    0.500000       NaN     0.700000  
std     0.707107       NaN     0.948683  
min     0.000000       NaN     0.000000  
25%     0.000000       NaN     0.000000  
50%     0.000000       NaN     0.000000  
75%     1.000000       NaN     1.750000  
max     2.000000       NaN     2.000000

In [148]:
#Check detail dan nilai NaN pada data_uji. gunakan print untuk Nilai NaN
#expected result

Age             0
Cabin          10
Embarked        0
Fare            0
Name            0
Parch           0
PassengerId     0
Pclass          0
Sex             0
SibSp           0
Survived       10
Ticket          0
Title           0
Family_Size     0
dtype: int64


Age  Cabin       Fare      Parch  PassengerId     Pclass  \
count  10.000000    0.0  10.000000  10.000000     10.00000  10.000000   
mean   30.150000    NaN  12.270010   0.200000    896.50000   2.800000   
std    14.541225    NaN   7.777627   0.421637      3.02765   0.421637   
min    14.000000    NaN   7.000000   0.000000    892.00000   2.000000   
25%    21.250000    NaN   7.679200   0.000000    894.25000   3.000000   
50%    26.500000    NaN   8.943750   0.000000    896.50000   3.000000   
75%    33.375000    NaN  11.637500   0.000000    898.75000   3.000000   
max    62.000000    NaN  29.000000   1.000000    901.00000   3.000000   

           SibSp  Survived  Family_Size  
count  10.000000       0.0    10.000000  
mean    0.500000       NaN     0.700000  
std     0.707107       NaN     0.948683  
min     0.000000       NaN     0.000000  
25%     0.000000       NaN     0.000000  
50%     0.000000       NaN     0.000000  
75%     1.000000       NaN     1.750000  
max     2.000000       NaN     2.000000

Terlihat untuk data cabin memiliki banyak nilai NaN. Jadi akan kita drop data tersebut. Name, Ticket, Title juga akan kita drop karena memiliki banyak nilai unik dan berbentuk string.

In [9]:
#drop kolom yang tidak dibutuhkan dari kedua dataframe, yaitu kolom cabin, name, ticket, dan title
data_mentah = data_mentah.drop(columns = ['Cabin','Name','Ticket','Title'], errors = 'ignore')
data_uji = data_uji.drop(columns = ['Cabin','Name','Ticket','Title'], errors = 'ignore')

In [10]:
# tampilkan head mentah
display(data_mentah.head())

Age Embarked     Fare  Parch  PassengerId  Pclass     Sex  SibSp  \
0  22.0        S   7.2500      0            1       3    male      1   
1  38.0        C  71.2833      0            2       1  female      1   
2  26.0        S   7.9250      0            3       3  female      0   
3  35.0        S  53.1000      0            4       1  female      1   
4  35.0        S   8.0500      0            5       3    male      0   

   Survived  Family_Size  
0       0.0            1  
1       1.0            1  
2       1.0            0  
3       1.0            1  
4       0.0            0

In [150]:
#tampilkan head mentah
#expected result

Age Embarked     Fare  Parch  PassengerId  Pclass     Sex  SibSp  \
0  22.0        S   7.2500      0            1       3    male      1   
1  38.0        C  71.2833      0            2       1  female      1   
2  26.0        S   7.9250      0            3       3  female      0   
3  35.0        S  53.1000      0            4       1  female      1   
4  35.0        S   8.0500      0            5       3    male      0   

   Survived  Family_Size  
0       0.0            1  
1       1.0            1  
2       1.0            0  
3       1.0            1  
4       0.0            0

In [11]:
# tampilkan head uji
display(data_uji.head())

Age Embarked     Fare  Parch  PassengerId  Pclass     Sex  SibSp  \
0  34.5        Q   7.8292      0          892       3    male      0   
1  47.0        S   7.0000      0          893       3  female      1   
2  62.0        Q   9.6875      0          894       2    male      0   
3  27.0        S   8.6625      0          895       3    male      0   
4  22.0        S  12.2875      1          896       3  female      1   

   Survived  Family_Size  
0       NaN            0  
1       NaN            1  
2       NaN            0  
3       NaN            0  
4       NaN            2

In [151]:
#tampilkan head uji
#expected result

Age Embarked     Fare  Parch  PassengerId  Pclass     Sex  SibSp  \
0  34.5        Q   7.8292      0          892       3    male      0   
1  47.0        S   7.0000      0          893       3  female      1   
2  62.0        Q   9.6875      0          894       2    male      0   
3  27.0        S   8.6625      0          895       3    male      0   
4  22.0        S  12.2875      1          896       3  female      1   

   Survived  Family_Size  
0       NaN            0  
1       NaN            1  
2       NaN            0  
3       NaN            0  
4       NaN            2

In [12]:
# untuk data mentah, mengubah male : 0, female : 1, kolom embarked S = 1, C = 2, Q = 3
data_mentah['Sex'].replace({'female' : 1,'male' : 0}, inplace=True)
data_mentah['Embarked'].replace({'S' : 1, 'C' : 2, 'Q' : 3}, inplace=True)
display(data_mentah.head())

Age  Embarked     Fare  Parch  PassengerId  Pclass  Sex  SibSp  Survived  \
0  22.0         1   7.2500      0            1       3    0      1       0.0   
1  38.0         2  71.2833      0            2       1    1      1       1.0   
2  26.0         1   7.9250      0            3       3    1      0       1.0   
3  35.0         1  53.1000      0            4       1    1      1       1.0   
4  35.0         1   8.0500      0            5       3    0      0       0.0   

   Family_Size  
0            1  
1            1  
2            0  
3            1  
4            0

In [152]:
#sekarang kita akan mengubah male : 0 dan female menjadi 1 pada kolom Sex. S = 1, C = 2, Q = 3 pada kolom embarked. Ubah kedua dataframe. 
#tampilkan head pada data mentah
#manfaatkan modul replace

#expected result

Age  Embarked     Fare  Parch  PassengerId  Pclass  Sex  SibSp  Survived  \
0  22.0         1   7.2500      0            1       3    0      1       0.0   
1  38.0         2  71.2833      0            2       1    1      1       1.0   
2  26.0         1   7.9250      0            3       3    1      0       1.0   
3  35.0         1  53.1000      0            4       1    1      1       1.0   
4  35.0         1   8.0500      0            5       3    0      0       0.0   

   Family_Size  
0            1  
1            1  
2            0  
3            1  
4            0

Sekarang kita akan membagi data_mentah menjadi data training dan test


In [13]:
# menggunakan data mentah dan mendefinisikan variabel matrix X dan y
X = data_mentah.loc[:, data_mentah.columns != 'Survived']
y = data_mentah.loc[:, data_mentah.columns == 'Survived']

In [14]:
X.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Age          891 non-null    float64
 1   Embarked     891 non-null    int64  
 2   Fare         891 non-null    float64
 3   Parch        891 non-null    int64  
 4   PassengerId  891 non-null    int64  
 5   Pclass       891 non-null    int64  
 6   Sex          891 non-null    int64  
 7   SibSp        891 non-null    int64  
 8   Family_Size  891 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 62.8 KB


In [153]:
#pilih data yang akan menjadi fitur dan label. gunakan data_mentah karena kita akan membuat model
X = None
y = None

In [15]:
# create the model dari variabel matrix X dan y
model_lr = LogisticRegression()
rfe = fs.RFE(model_lr)
rfe.fit(X,y)
print(f'Support = {rfe.support_}')
print(f'Ranking = {rfe.ranking_}')

Support = [False  True False False False  True  True  True False]
Ranking = [4 1 5 3 6 1 1 1 2]


In [154]:
#import sklearn.feature_selection pada sel package/library aliaskan sebagai fs
#import module LogisticRegression dari sklearn.linear_model. taruh pada sel package/library

#expected result

model_lr = LogisticRegression()
rfe = fs.RFE(model_ls)
rfe.fit(X,y)
print(f'Support = {rfe.support_}')
print(f'Ranking = {rfe.ranking_}')

Support = [False  True False False False  True  True  True False]
Ranking = [4 1 5 3 6 1 1 1 2]


C:\Users\thiop\Envs\public_op\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
C:\Users\thiop\Envs\public_op\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression

In [16]:
# mengubah nilai X dengan mengikuti nilai kolom terbaik, memilih kolom yang ranking 1
X = data_mentah[['Embarked','Pclass','Sex','SibSp']]


In [155]:
#sekarang kita ubah nilai X dengan mengikuti nilai kolom terbaik. urutan pada list ranking sesuai urutan pada saat inisiasi variabel X
# pilih ranking 1
# expected result

X = data_mentah[['']]

In [17]:
# membagi data menjadi training dan testing dengan test_size = 0.2
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=123)

print(f'dimensi dari X_train : {len(X_train)}')
print(f'dimensi dari y_train : {len(y_train)}')
print(f'dimensi dari X_test : {len(X_test)}')
print(f'dimensi dari y_test : {len(y_test)}')

dimensi dari X_train : 712
dimensi dari y_train : 712
dimensi dari X_test : 179
dimensi dari y_test : 179


In [156]:
'''import/tambahkan semua package/library pada sel paling atas(terkait library)
masukkan kedua kolom tersebut pada variabel yang bersesuaian.
kita bagi menjadi data training dan test dengan test_size = 0.2
'''

#expected result

print(f'dimensi dari X_train : {len(X_train)}')
print(f'dimensi dari y_train : {len(y_train)}')
print(f'dimensi dari X_test : {len(X_test)}')
print(f'dimensi dari y_test : {len(y_test)}')

dimensi dari X_train : 712
dimensi dari y_train : 712
dimensi dari X_test : 179
dimensi dari y_test : 179


4. Data Processing

In [18]:
# training model
# menggunakan object logistic regression yang sudah dibuat sebelumnya

model = LogisticRegression()

# fit the model with data
print(model.fit(X_train,y_train))

LogisticRegression()


In [157]:
#training model
#langsung gunakan object logistic regression yang sudah dibuat sebelumnya

# fit the model with data


LogisticRegression()

In [19]:
# evaluasi model yang sudah diperoleh
y_predict = model.predict(X_test)

# menghitung nilai auc sebagai indikator apakah modelnya fit atau tidak
auc = metrics.accuracy_score(y_test,y_predict)
print(f'Nilai AUC : {auc}')

Nilai AUC : 0.8156424581005587


In [158]:
#Evaluasi
#kita buat variabel y_predict
#import modul metrics dari sklearn. Taruh pada sel package/library di awal
#test nilai AUC

#expected result
print(f'Nilai AUC : {auc}')

Nilai AUC : 0.8156424581005587


Sekarang kita akan mencoba mengklasifikasikan data_uji kita

In [20]:
# poin 1.
# membuat sebuah variabel untuk menampung kolom dari data_uji yang kolomnya sama dengan inisiasi variabel X
data_uji['Sex'].replace({'female' : 1,'male' : 0}, inplace=True)
data_uji['Embarked'].replace({'S' : 1, 'C' : 2, 'Q' : 3}, inplace=True)

X_baru = data_uji[['Embarked','Pclass','Sex','SibSp']]

In [21]:
# poin 2.
# melakukan prediksi dengan menggunakan variabel pada poin satu
y_predict_baru = model.predict(X_baru)

In [22]:
# poin 3.
# memasukkan variabel pada poin dua pada data uji kolom Survived
data_uji['Survived'] = y_predict_baru

In [23]:
# poin 4.
# menampilkan data uji
display(data_uji)

Age  Embarked     Fare  Parch  PassengerId  Pclass  Sex  SibSp  Survived  \
0  34.5         3   7.8292      0          892       3    0      0       0.0   
1  47.0         1   7.0000      0          893       3    1      1       1.0   
2  62.0         3   9.6875      0          894       2    0      0       0.0   
3  27.0         1   8.6625      0          895       3    0      0       0.0   
4  22.0         1  12.2875      1          896       3    1      1       1.0   
5  14.0         1   9.2250      0          897       3    0      0       0.0   
6  30.0         3   7.6292      0          898       3    1      0       1.0   
7  26.0         1  29.0000      1          899       2    0      1       0.0   
8  18.0         2   7.2292      0          900       3    1      0       1.0   
9  21.0         1  24.1500      0          901       3    0      2       0.0   

   Family_Size  
0            0  
1            1  
2            0  
3            0  
4            2  
5            0  
6            0  
7            2  
8            0  
9            2

In [159]:
#buat sebuah variabel untuk menampung kolom dari data_uji. Ingat kolom yang dipilih harus sama dengan semua kolom pada saat memilih X
#lakukan prediksi dengan menggunakan variabel pada poin satu.
#masukkan variabel pada poin dua pada data uji kolom Survived
#tampilkan data_uji

#expected result

Age  Embarked     Fare  Parch  PassengerId  Pclass  Sex  SibSp  Survived  \
0  34.5         3   7.8292      0          892       3    0      0       0.0   
1  47.0         1   7.0000      0          893       3    1      1       1.0   
2  62.0         3   9.6875      0          894       2    0      0       0.0   
3  27.0         1   8.6625      0          895       3    0      0       0.0   
4  22.0         1  12.2875      1          896       3    1      1       1.0   
5  14.0         1   9.2250      0          897       3    0      0       0.0   
6  30.0         3   7.6292      0          898       3    1      0       1.0   
7  26.0         1  29.0000      1          899       2    0      1       0.0   
8  18.0         2   7.2292      0          900       3    1      0       1.0   
9  21.0         1  24.1500      0          901       3    0      2       0.0   

   Family_Size  
0            0  
1            1  
2            0  
3            0  
4            2  
5            0  
6            0  
7            2  
8            0  
9            2

In [160]:
#tuliskan hasil analisa kalian disini

In [30]:
display(data_uji.groupby('Survived', as_index = False)['Sex'].count())

Survived  Sex
0       0.0    6
1       1.0    4

dari hasil analisis di atas menggunakan logistic regression, dapat diperoleh bahwa :

1. dalam hal ini, belum bisa diperoleh kesimpulan secara statistik, dikarenakan masih belum terdapat koefisien dari masing-masing variabel prediktor.

2. sehingga kesimpulan yang dapat diperoleh hanya berupa deskriptif saja, seperti :

- apabila score 1 menyatakan survive dan 0 menyatakan selain survive, maka penumpang yang berusia 47 tahun dengan embarked 1 yang memiliki poin fare sebesar 7 dan poin parch sebesar 0 dengan passenger id 893 yang memiliki Pclass 3 dengan jenis kelamin perempuan dan sibspnya 1 dengan family sizenya 1, cenderung dinyatakan survive.

- penumpang yang berusia 22 tahun dengan embarked 1 yang memiliki poin fare sebesar 12.2875 dan poin parch sebesar 1 dengan passenger id 896 yang memiliki Pclass 3 dengan jenis kelamin perempuan dan sibspnya 1 dengan family sizenya 2, cenderung dinyatakan survive.

- penumpang yang berusia 30 tahun dengan embarked 3 yang memiliki poin fare sebesar 7.6292 dan poin parch sebesar 0 dengan passenger id 898 yang memiliki Pclass 3 dengan jenis kelamin perempuan dan sibspnya 0 dengan family sizenya 0, cenderung dinyatakan survive.

- penumpang yang berusia 18 tahun dengan embarked 2 yang memiliki poin fare sebesar 7.2292 dan poin parch sebesar 0 dengan passenger id 900 yang memiliki Pclass 3 dengan jenis kelamin perempuan dan sibspnya 0 dengan family sizenya 0, cenderung dinyatakan survive.

- sisa dari data tersebut dinyatakan selain survive.

- terdapat 4 penumpang yang berjenis kelamin perempuan cenderung dinyatakan survive dibandingkan dengan 6 penumpang yang berjenis kelamin laki-laki.

- nilai auc yang diperoleh dari data mentah adalah 0.8156 atau 81.56%. AUC (Area Under Curve) sendiri merepresentasikan evaluasi dari hasil klasifikasi. semakin tinggi nilai auc, maka dapat dikatakan modelnya semakin baik.